subsys/bluetooth/services/ancs_app_attr_get.c

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2016 - 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+/* Disclaimer: This client implementation of the Apple Notification Center Service can and will be changed at any time by Nordic Semiconductor ASA.
+ * Server implementations such as the ones found in iOS can be changed at any time by Apple and may cause this client implementation to stop working.
+ */
+
+#include <bluetooth/services/ancs_c.h>
+#include <bluetooth/services/ancs_app_attr_get.h>
+#include <logging/log.h>
+#include "ancs_c_internal.h"
+
+LOG_MODULE_DECLARE(ancs_c, CONFIG_BT_GATT_ANCS_C_LOG_LEVEL);
+
+/** Maximum length of the data that can be sent in a write. */
+#define ANCS_GATTC_WRITE_PAYLOAD_LEN_MAX CONFIG_BT_GATT_ANCS_C_CP_BUFF_SIZE
+
+/**@brief Enumeration for keeping track of the state-based encoding while requesting app attributes. */
+typedef enum {
+	/** Currently encoding the command ID. */
+	APP_ATTR_COMMAND_ID,
+	/** Currently encoding the app ID. */
+	APP_ATTR_APP_ID,
+	/** Currently encoding the attribute ID. */
+	APP_ATTR_ATTR_ID,
+	/** Encoding done. */
+	APP_ATTR_DONE,
+	/** Encoding aborted. */
+	APP_ATTR_ABORT
+} encode_app_attr_t;
+
+/**@brief Function for determining whether an attribute is requested.
+ *
+ * @param[in] ancs_c iOS notification structure. This structure must be supplied by
+ *                   the application. It identifies the particular client instance to use.
+ *
+ * @return True  If it is requested
+ * @return False If it is not requested.
+*/
+static bool app_attr_is_requested(struct bt_gatt_ancs_c *ancs_c,
+				  uint32_t attr_id)
+{
+	if (ancs_c->ancs_app_attr_list[attr_id].get == true) {
+		return true;
+	}
+	return false;
+}
+
+/**@brief Function for counting the number of attributes that will be requested upon a "get app attributes" command.
+ *
+ * @param[in] ancs_c iOS notification structure. This structure must be supplied by
+ *                   the application. It identifies the particular client instance to use.
+ *
+ * @return           Number of attributes that will be requested upon a "get app attributes" command.
+*/
+static uint32_t app_attr_nb_to_get(struct bt_gatt_ancs_c *ancs_c)
+{
+	uint32_t attr_nb_to_get = 0;
+	for (uint32_t i = 0; i < (sizeof(ancs_c->ancs_app_attr_list) /
```

Comment:
```
sizeof(ancs_c->ancs_app_attr_list) /
				  sizeof(bt_gatt_ancs_c_attr_list_t)
```
can be change to `ARRAY_SIZE(ancs_c->ancs_app_attr_list)` ?

Reply:
Done